src/app/layout/qgslayoutdesignerdialog.cpp

```diff
@@ -1523,6 +1759,105 @@ void QgsLayoutDesignerDialog::initializeRegistry()
 
 }
 
+bool QgsLayoutDesignerDialog::containsWmsLayers() const
+{
+  QList< QgsLayoutItemMap *> maps;
+  mLayout->layoutItems( maps );
+
+  for ( QgsLayoutItemMap *map : qgis::as_const( maps ) )
+  {
+    if ( map->containsWmsLayer() )
+      return true;
+  }
+  return false;
+}
+
+void QgsLayoutDesignerDialog::showWmsPrintingWarning()
+{
+  QgsSettings settings;
+  bool displayWMSWarning = settings.value( QStringLiteral( "/UI/displayComposerWMSWarning" ), true ).toBool();
+  if ( displayWMSWarning )
+  {
+    QgsMessageViewer *m = new QgsMessageViewer( this );
+    m->setWindowTitle( tr( "Project Contains WMS Layers" ) );
+    m->setMessage( tr( "Some WMS servers (e.g. UMN mapserver) have a limit for the WIDTH and HEIGHT parameter. Printing layers from such servers may exceed this limit. If this is the case, the WMS layer will not be printed" ), QgsMessageOutput::MessageText );
+    m->setCheckBoxText( tr( "Don't show this message again" ) );
+    m->setCheckBoxState( Qt::Unchecked );
+    m->setCheckBoxVisible( true );
+    m->setCheckBoxQgsSettingsLabel( QStringLiteral( "/UI/displayComposerWMSWarning" ) );
+    m->exec(); //deleted on close
+  }
+}
+
+bool QgsLayoutDesignerDialog::requiresRasterization() const
+{
+  QList< QgsLayoutItem *> items;
+  mLayout->layoutItems( items );
+
+  for ( QgsLayoutItem *currentItem : qgis::as_const( items ) )
+  {
+    if ( currentItem->requiresRasterization() )
+      return true;
+  }
+  return false;
+}
+
+bool QgsLayoutDesignerDialog::containsAdvancedEffects() const
+{
+  QList< QgsLayoutItem *> items;
+  mLayout->layoutItems( items );
+
+  for ( QgsLayoutItem *currentItem : qgis::as_const( items ) )
+  {
+    if ( currentItem->containsAdvancedEffects() )
+      return true;
+  }
+  return false;
+}
+
+void QgsLayoutDesignerDialog::showRasterizationWarning()
+{
+
+  if ( mLayout->customProperty( QStringLiteral( "rasterise" ), false ).toBool() ||
+       mLayout->customProperty( QStringLiteral( "forceVector" ), false ).toBool() )
+    return;
+
+  QgsMessageViewer *m = new QgsMessageViewer( this, QgsGuiUtils::ModalDialogFlags, false );
+  m->setWindowTitle( tr( "Composition Effects" ) );
+  m->setMessage( tr( "Advanced composition effects such as blend modes or vector layer transparency are enabled in this layout, which cannot be printed as vectors. Printing as a raster is recommended." ), QgsMessageOutput::MessageText );
+  m->setCheckBoxText( tr( "Print as raster" ) );
+  m->setCheckBoxState( Qt::Checked );
+  m->setCheckBoxVisible( true );
+  m->showMessage( true );
+
+  mLayout->setCustomProperty( QStringLiteral( "rasterise" ), m->checkBoxState() == Qt::Checked );
+  //make sure print as raster checkbox is updated
+  mLayoutPropertiesWidget->updateGui();
+
+  delete m;
```

Comment:
Do you think that a `unique_ptr` could used in this case?

Reply:
Yes - actually doesn't need to be on the heap here at all.

Reply:
Fixed